common/TelemetryEvents/TelemetryResult.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Windows.DevHome.SDK;
+
+namespace DevHome.Common.TelemetryEvents;
+
+public class TelemetryResult
+{
+    private const int SuccessHResult = 0;
+
+    private const int InvalidArgHResult = unchecked((int)0x80070057);
+
+    public int HResult { get; private set; }
+
+    public ProviderOperationStatus Status { get; private set; }
+
+    public string? DisplayMessage { get; private set; }
+
+    public string? DiagnosticText { get; private set; }
+
+    public TelemetryResult(ProviderOperationResult? result)
+    {
+        UpdateProperties(result);
+    }
+
+    public TelemetryResult(int hResult, string displayMessage, string diagnosticText)

```

Comment:
This seems like a good candidate to document in https://github.com/microsoft/devhome/tree/main/docs/tools/common. That way we don't forget things, and it encourages reuse.

Reply:
will do. Thanks

Reply:
updated.